modules/markup/orgmode/orgmode.go

```diff
@@ -38,6 +40,28 @@ func (Parser) Extensions() []string {
 // Render renders orgmode rawbytes to HTML
 func Render(rawBytes []byte, urlPrefix string, metas map[string]string, isWiki bool) []byte {
 	htmlWriter := org.NewHTMLWriter()
+	htmlWriter.HighlightCodeBlock = func(source, lang string, inline bool) string {
+		var w strings.Builder
+		if _, err := w.WriteString(`<pre>`); err != nil {
+			return ""
+		}
+
+		// include language-x class as part of commonmark spec
+		if _, err := w.WriteString(`<code class="chroma language-` + string(lang) + `">`); err != nil {
+			return ""
+		}
+
+		lexer := lexers.Get(lang)
+		if _, err := w.WriteString(highlight.Code(lexer.Config().Filenames[0], source)); err != nil {
+			return ""
+		}
+
+		if _, err := w.WriteString("</code></pre>"); err != nil {
+			return ""
+		}
+
+		return w.String()
+	}
```

Comment:
Most of this is duplicating with `modules/markup/markdown/markdown.go`, can we extract it to a shared function like `RenderCodePre`?

Reply:
I think they are similar but different functions. `markdown`'s will be invoked twice, once before rendering the code, another after rendering the code. But orgmode is only one function.
Is it really valuable to merge the two functions since Gitea only supports the two formats.

Reply:
I think this looks sufficiently different from the code in `markdown.go`